src/geo/transform.js

```diff
@@ -1071,29 +1138,58 @@ class Transform {
         }
     }
 
+    calculatePosMatrix(unwrappedTileID: UnwrappedTileID, worldSize: number): Float32Array {
+        const canonical = unwrappedTileID.canonical;
+        const scale = worldSize / this.zoomScale(canonical.z);
+        const unwrappedX = canonical.x + Math.pow(2, canonical.z) * unwrappedTileID.wrap;
+
+        const posMatrix = mat4.identity(new Float64Array(16));
+        mat4.translate(posMatrix, posMatrix, [unwrappedX * scale, canonical.y * scale, 0]);
+        mat4.scale(posMatrix, posMatrix, [scale / EXTENT, scale / EXTENT, 1]);
+
+        return posMatrix;
+    }
+
     /**
-     * Calculate the posMatrix that, given a tile coordinate, would be used to display the tile on a map.
+     * Calculate the fogTileMatrix that, given a tile coordinate, can be used to
+     * calculate its position relative to the camera in units of pixels divided
+     * by the map height. Used with fog for consistent computation of distance
+     * from camera.
+     *
      * @param {UnwrappedTileID} unwrappedTileID;
      * @private
      */
-    calculatePosMatrix(unwrappedTileID: UnwrappedTileID, aligned: boolean = false): Float32Array {
-        const posMatrixKey = unwrappedTileID.key;
-        const cache = aligned ? this._alignedPosMatrixCache : this._posMatrixCache;
-        if (cache[posMatrixKey]) {
-            return cache[posMatrixKey];
+    calculateFogTileMatrix(unwrappedTileID: UnwrappedTileID): Float32Array {
+        const fogTileMatrixKey = unwrappedTileID.key;
+        const cache = this._fogTileMatrixCache;
+        if (cache[fogTileMatrixKey]) {
+            return cache[fogTileMatrixKey];
         }
 
-        const canonical = unwrappedTileID.canonical;
-        const scale = this.worldSize / this.zoomScale(canonical.z);
-        const unwrappedX = canonical.x + Math.pow(2, canonical.z) * unwrappedTileID.wrap;
+        const posMatrix = this.calculatePosMatrix(unwrappedTileID, this.cameraWorldSize);
+        mat4.multiply(posMatrix, this.worldToFogMatrix, posMatrix);
 
-        const posMatrix = mat4.identity(new Float64Array(16));
-        mat4.translate(posMatrix, posMatrix, [unwrappedX * scale, canonical.y * scale, 0]);
-        mat4.scale(posMatrix, posMatrix, [scale / EXTENT, scale / EXTENT, 1]);
+        cache[fogTileMatrixKey] = new Float32Array(posMatrix);
+        return cache[fogTileMatrixKey];
+    }
+
+    /**
+     * Calculate the projMatrix that, given a tile coordinate, would be used to display the tile on the screen.
+     * @param {UnwrappedTileID} unwrappedTileID;
+     * @private
+     */
+    calculateProjMatrix(unwrappedTileID: UnwrappedTileID, aligned: boolean = false): Float32Array {
```

Comment:
After the rename `projMatrix` is used to refer to both the world -> gl coordinate matrix (`transform.projMatrix`) and to the tile -> gl coordinate matrix returned by this method. I'm not sure what the best naming here would be but maybe these two things should have different names

Reply:
I did make this change to reflect exactly that, `transform.projMatrix` for pixel -> gl and `tile.projMatrix` for tile -> gl.
The previous `posMatrix` got confusing because we had to add the separate `posMatrix` for tile -> pixel.